[RHACS] OCP OAuth integration (#38980)
openshift/openshift-docs: operating/manage-user-access/configure-ocp-oauth.adoc

[id="configure-ocp-oauth"]
= Configuring {ocp} OAuth server as an identity provider
include::modules/common-attributes.adoc[]
:context: configure-ocp-oauth

toc::[]

[role="_abstract"]

> {ocp} includes a built-in OAuth server and you can use it as an authentication provider for {product-title} (RHACS).

Reviewer: This is only applicable when you install using roxctl on OpenShift. This is never applicable if you are using helm or the operator.  The only time you would ever follow the procedure is when you use roxctl on OpenShift to install (which is highly unlikely.) This should go in the upgrade guide and apply only to OpenShift. I added the comments to the upgrade PR.

[IMPORTANT]
====

> * Integration with {ocp} OAuth server is only available if you install the {product-title} on {ocp}. If you have installed {product-title} on Kubernetes, the user interface (UI) shows the option to configure {ocp} OAuth server, but the integration will not work.

Reviewer: This is not applicable to Kubernetes in any way.  Theres no need to reference Kubernetes at all.

* To disable the {ocp} OAuth server integration, set the `ROX_ENABLE_OPENSHIFT_AUTH` environment variable as `false` in Central:
+
[source,terminal]
----
$ oc -n stackrox set env deploy/central ROX_ENABLE_OPENSHIFT_AUTH=false <1>
----

> <1> If you use Kubernetes, enter `kubectl` instead of `oc`.

Reviewer: This will never apply to Kubernetes.

+
When you disable the {ocp} OAuth server integration, all the previously configured {ocp} OAuth providers stop working and they are not listed as a login option. The UI still shows the option to configure {ocp} OAuth server, but the integration will not work.
* Integration with {ocp} OAuth server is only available for {ocp} 4.1 and newer.

> * If you use a custom TLS certificate for {ocp} OAuth server, you must add the certificate as a trusted root CA or as a Kubernetes secret `/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt`. Otherwise, Central will not establish a TLS connection with the {ocp} OAuth server.

Reviewer: If it is not already trusted. If they are using custom certs is likely they will have already trusted this CA for other configuration.

[id="configure-ocp-oauth"]
= Configuring {ocp} OAuth server as an identity provider
include::modules/common-attributes.adoc[]
:context: configure-ocp-oauth

toc::[]

[role="_abstract"]

> {ocp} includes a built-in OAuth server and you can use it as an authentication provider for {product-title} (RHACS).

Reviewer 2: ```suggestion
{ocp} includes a built-in OAuth server that you can use as an authentication provider for {product-title} (RHACS).
```

[CAUTION]
====

> The {ocp} OAuth server integration only works when you install {product-title} by using the `roxctl` CLI on {ocp}. It doesn't work when you install {product-title} by using Helm or the {product-title} Operator.

Reviewer 2: ```suggestion
The {ocp} OAuth server integration only works when you install {product-title} by using the `roxctl` CLI on {ocp}. It does not work when you install {product-title} by using Helm or the {product-title} Operator.
```

====

[IMPORTANT]
====

> * To disable the {ocp} OAuth server integration, set the `ROX_ENABLE_OPENSHIFT_AUTH` environment variable as `false` in Central:

Reviewer 2: ```suggestion
* To disable the {ocp} OAuth server integration, set the `ROX_ENABLE_OPENSHIFT_AUTH` environment variable to `false` in Central:
```

+
[source,terminal]
----
$ oc -n stackrox set env deploy/central ROX_ENABLE_OPENSHIFT_AUTH=false
----
+
When you disable the {ocp} OAuth server integration, all the previously configured {ocp} OAuth providers stop working and they are not listed as a login option. The UI still shows the option to configure {ocp} OAuth server, but the integration will not work.
* Integration with {ocp} OAuth server is only available for {ocp} 4.1 and newer.
* If you use a custom TLS certificate for {ocp} OAuth server, you must add the certificate as a trusted root CA or as a Kubernetes secret `/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt`. Otherwise, Central will not establish a TLS connection with the {ocp} OAuth server.

> ====

Reviewer 2: In general, we try to avoid stacking admonitions. I am guessing there's not an easy way out of this one, though.
